torch/csrc/lazy/python/init.cpp

```diff
@@ -1,12 +1,86 @@
 #include <torch/csrc/lazy/python/init.h>
 
+#include <c10/core/Device.h>
+#include <torch/csrc/jit/python/pybind.h>
+#include <torch/csrc/lazy/backend/backend_device.h>
 #include <torch/csrc/lazy/core/debug_util.h>
+#include <torch/csrc/lazy/core/lazy_graph_executor.h>
+#include <torch/csrc/lazy/core/metrics.h>
 #include <torch/csrc/lazy/python/python_util.h>
+#include <torch/csrc/lazy/ts_backend/ts_backend_impl.h>
+#include <string>
+#include <vector>
 
 namespace torch {
 namespace lazy {
 
-void initLazyBindings(PyObject* /* module */){
+
+struct NoGilSection {
+  NoGilSection() : state(PyEval_SaveThread()) {}
+  ~NoGilSection() { PyEval_RestoreThread(state); }
+  PyThreadState* state = nullptr;
+};
+
+torch::lazy::BackendDevice GetDeviceOrCurrent(const std::string& device_str) {
+  if (device_str.empty()) {
+    return torch::lazy::BackendDevice();
+  }
+  return torch::lazy::atenDeviceToBackendDevice(c10::Device(device_str));
+}
+
+void StepMarker(const std::string& device_str,
+                const std::vector<std::string>& devices, bool wait) {
+  auto device = GetDeviceOrCurrent(device_str);
+  torch::lazy::LazyGraphExecutor::Get()->SyncLiveTensorsGraph(&device, devices, wait);
+  torch::lazy::LazyGraphExecutor::Get()->MarkStep(device);
+
+  // TODO(whc) enable or delete this? (have we used it?)
```

Comment:
As we land these bindings, i'm wondering if we should try to simplify by not having helper functions like 'StepMarker'.

Taking the full view of the mark step API flow, it looks like

1 torch._lazy.mark_step (python code in torch/_lazy/__init__.py)
2   -> torch._C._lazy._mark_step (binding code in lazy/python/init.cpp)
3      -> StepMarker (local helper function in lazy/python/init.cpp)
4        -> finally calls torch::lazy::blah APIs

(2) can always be squished into (3) but that can make gdb debugging more annoying.  But I also wonder if we can simplify to only have 3 layers, and generally flush stuff that would be in (2) either up into python in (1) or down into the core APIs (4).

Thoughts? @Krovatkin @alanwaketan

Reply:
I think this is just Google's coding style to prefer wrapping a pile of code into a function just for readability not reusability.

Reply:
I'm fine either way. 

> (2) can always be squished into (3) but that can make gdb debugging more annoying.

If users don't really need a python binding for 2), I'd suggest we go for it.